OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -0,0 +1,641 @@
+ * give to the state variables. You probably want to use the trajectory with an
+ * OpenSim::Model, through which the state variables have a meaning (e.g.,
+ * `model.getStateVariableValue(states[0], "soleus_r/activation")`).
+ *
+ * SimTK::State%s have a tight association with a specific OpenSim::Model
+ * (actually, with the SimTK::System within an OpenSim::Model). However,
+ * neither the StatesTrajectory nor the OSTATES file knows the model to which
+ * it corresponds. So, for example, you could use a single StatesTrajectory
+ * with a generic gait2392 model as well as with a scaled (subject-specific)
+ * gait2392 model. This flexibility may be beneficial in some scenarios, but
+ * also allows one to accidentally use the wrong model with a given states
+ * trajectory, potentially leading to silent errors that could compromise a
+ * scientific study.
+ *
+ * To increase your confidence that a StatesTrajectory matches a given Model,
+ * you can perform some weak checks with isCompatibleWith().
+ *
+ * TODO acceleration-level calculations?
+ *
+ * ### File format
+ * StatesTrajectory files use the file extension `.OSTATES`, with the XML
+ * format. Therefore, you could theoretically edit a StatesTrajectory file in
+ * a typical text editing program, or in Python/MATLAB using XML libraries
+ * (such a process is is likely to be painful; consider using the
+ * createFromStatesStorage() utility instead).
+ *
+ * A SimTK::State object contains many different types of data, but only some
+ * are saved into the OSTATES file:
+ * 
+ * type of data                 | saved in OSTATES?
+ * ---------------------------- | -------------
+ * (continuous) state variables | yes
+ * discrete state variables     | yes
+ * modeling options             | yes
+ * cache variables              | no
+ *
+ * The cache variables (e.g., total system mass, control signals) are not saved
+ * to the OSTATES file because they can be regenerated from the other data and
+ * the model.
+ *
+ * %OpenSim Object%s (Model OSIM files, Tool setup files) also use an
+ * XML format, but that format is *completely unrelated* to the XML format used
+ * for OSTATES files.
+ *
+ * ### Usage
+ * Here are a few basic things you can do with a StatesTrajectory, assuming you
+ * already have one:
+ * @code{.cpp}
+ * StatesTrajectory states = getStatesTrajectorySomehow();
+ * const double numStates = states.getSize();
+ * const double initialTime = states[0].getTime();
+ * const double finalTime = states.back().getTime();
+ * @endcode
+ *
+ * Without a model, you can access the state variable values, but you won't
+ * know the identity of such state variables.
+ * @code{.cpp}
+ * int numGeneralizedCoordinates = states[0].getNQ();
+ * const SimTK::Vector& generalizedCoordinateValues = states[0].getQ();
+ * @endcode
+ *
+ * To do most things with the StatesTrajectory, you'll need a model. **It is
+ * essential that you've called `Model::initSystem()` before you try to use any
```

Comment:
I would prefer to leave this part of the documentation as-is. Two reasons: (1) you probably need more than just the model and system; you also need some basic realization (model, instance); (2) I believe that most users don't know what the "system" is, and don't know that `initSystem()` creates  a `SimTK::System` (I think they view "system" as just a generic term).